dlib/pixel.h

```diff
@@ -942,11 +1032,12 @@ namespace dlib
 
     // -----------------------------
 
-        struct HSL
+        // Helper struct used for HSL and HSV
+        struct HSX
```

Comment:
Everything looks good.  But why change this struct? It doesn't look like this is needed.  Like is there some generic programming I'm not seeing that benefits from this?  It seems like it would be better to have both an `HSL` and `HSV` struct type here.

Reply:
Ah, just to avoid another struct that would be exactly the same. Since this is not exposed to the user... but I can duplicate a HSV struct if you want.

Reply:
Yeah I mean just to keep the code and function contracts super clear what is what :+1: